### PR TITLE
[AKS] SL-1794 Additional action when changing joh for a session with multi part hearing via planner

### PR DESCRIPTION
**Pull request by zielinskikamil**
Added DragAndDropSession Action + unit test

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            .map(hpFact -> new FactMessage(RulesService.UPSERT_HEARING_PART, hpFact))
+            .collect(Collectors.toList());
+        facts.add(new FactMessage(RulesService.UPSERT_SESSION, sessionWithHpFacts.getSessionFact()));
+
+        return facts;
+    }
+
+    @Override
+    public List<UserTransactionData> generateUserTransactionData() {
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java]: There is currently UserTransactionDataPreparerService, so please use that.

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    public UUID getUserTransactionId() {
+        return dragAndDropSessionRequest.getUserTransactionId();
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        List<UUID> associatedEntitiesIds = new ArrayList<>();
+        associatedEntitiesIds.add(dragAndDropSessionRequest.getSessionId());
+        hearingParts.forEach(hp -> associatedEntitiesIds.add(hp.getId()));
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java]: Should we add hearings also?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        userTransactionDataList.add(new UserTransactionData("session",
+            session.getId(),
+            currentSessionAsString,
+            "update",
+            "update",
+            0)
+        );
+
+        for (HearingPart hp : hearingParts) {
+            String hearingPartAsString = null;
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java]: Here you can also use UserTransactionDataPreparerService.mapHearingPartToStrings, you can find an example in WithdrawHearingAction

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/SessionController.java [hunk 4]
```diff
@@ -138,8 +139,8 @@ public ResponseEntity insertSession(@Valid @RequestBody UpsertSession upsertSess
     }
 
     @PutMapping(path = "/update", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity updateSession(@RequestBody UpsertSession upsertSession) throws IOException {
-        UserTransaction ut = sessionService.updateSession(upsertSession);
+    public ResponseEntity updateSession(@RequestBody DragAndDropSessionRequest dropSessionRequest) throws IOException {
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/SessionController.java]: @ Valid annotation is probably needed here to actually validate the request.

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java (hunk 2 above)
Comment (zielinskikamil) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java]: in this action we are changing only session properties. 
There is a stroy to sync session start time with hearing part start time so within this task we could think about it

**Pull request merged**